android/services/mongodb-remote/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/remote/internal/SyncPerformanceTests.kt

```diff
@@ -0,0 +1,518 @@
+            "http://10.0.2.2:9090"
+        )
+    }
+
+    @Before
+    override fun setup() {
+        super.setup()
+
+        if (!Stitch.hasAppClient(stitchOutputAppName)) {
+            outputClient = Stitch.initializeAppClient(stitchOutputAppName)
+            outputClient.auth.loginWithCredential(AnonymousCredential())
+
+            outputColl = outputClient
+                .getServiceClient(RemoteMongoClient.factory, "mongodb-atlas")
+                .getDatabase(stitchOutputDbName)
+                .getCollection(stitchOutputCollName)
+        }
+    }
+
+    private fun setupIter() {
+        Assume.assumeTrue("no MongoDB URI in properties; skipping test", getMongoDbUri().isNotEmpty())
+        setup()
+
+        val app = createApp()
+
+        addProvider(app.second, ProviderConfigs.Anon)
+        mdbService = addService(
+                app.second,
+                "mongodb",
+                "mongodb1",
+                ServiceConfigs.Mongo(getMongoDbUri())).second
+
+        dbName = ObjectId().toHexString()
+        collName = ObjectId().toHexString()
+        namespace = MongoNamespace(dbName, collName)
+
+        val rule = RuleCreator.MongoDb(
+                database = dbName,
+                collection = collName,
+                roles = listOf(RuleCreator.MongoDb.Role(
+                        read = true, write = true
+                )),
+                schema = RuleCreator.MongoDb.Schema())
+        mdbRule = addRule(mdbService, rule)
+
+        this.client = getAppClient(app.first)
+        userId = Tasks.await(client.auth.loginWithCredential(AnonymousCredential())).id
+        mongoClient = client.getServiceClient(RemoteMongoClient.factory, "mongodb1")
+        coll = mongoClient.getDatabase(dbName).getCollection(collName)
+        (mongoClient as RemoteMongoClientImpl).dataSynchronizer.stop()
+        (mongoClient as RemoteMongoClientImpl).dataSynchronizer.disableSyncThread()
+        BaseStitchAndroidIntTest.testNetworkMonitor.connectedState = true
+    }
+
+    @After
+    override fun teardown() {
+        super.teardown()
+    }
+
+    private fun teardownIter() {
+        val syncedIds = Tasks.await(coll.sync().syncedIds)
+        Tasks.await(coll.sync().desyncMany(*syncedIds.toTypedArray()))
+        Tasks.await(coll.deleteMany(Document()))
+
+        if (::mongoClient.isInitialized) {
+            (mongoClient as RemoteMongoClientImpl).dataSynchronizer.close()
+            AndroidEmbeddedMongoClientFactory.getInstance().close()
+        }
+
+        teardown()
+    }
+
+    private fun runPerformanceTestWithParams(
+        testParams: TestParams,
+        block: () -> Unit
+    ) = runBlocking {
+
+        val runtime = Runtime.getRuntime()
+        stitchTestHost = testParams.stitchHostName
+
+        val resultId = ObjectId()
+        if (testParams.outputToStitch) {
+            val doc = testParams.toBson().append("_id", resultId)
+            Tasks.await(outputColl.insertOne(doc))
```

Comment:
My idea here was that we dont have to hold on to everything in memory and can just keep pushing the results into the results array in atlas as we get them (ie every combination of numDocs and docSize). I figured we dont care if the test fails because the results are still valid, but we can setup some kind of try {} catch{} that deleted the document at the end if you think thats better. Also happy to just make it one big insert but curious on your thoughts?

Reply:
That's fair. Yeah I'd be okay with having a failover here.